packages/projection/src/operators/Mappers/withHandles.ts

```diff
@@ -62,11 +67,45 @@ const tryCreateHandleOwnership = (
   }
 };
 
+const getHandleMetadata = (handleDataFields: Cardano.PlutusList, logger: Logger) =>
+  handleDataFields.items.map((data) => {
+    if (Cardano.util.isPlutusMap(data)) {
+      return Cardano.util.tryConvertPlutusMapToUtf8Record(data, logger);
+    }
+  });
```

Comment:
This is going to be constant (probably either 0th or 2nd element). No need to check all 3.
```suggestion
   const data = handleDataFields[2];
    if (Cardano.util.isPlutusMap(data)) {
      return Cardano.util.tryConvertPlutusMapToUtf8Record(data, logger);
    }
  });
```

Reply:
should be addressed: b2fc531cece813e1ea89db59d3ab90cfa6204bc2